.tekton/task.yaml

```
@@ -0,0 +1,167 @@
+          - 'describe cluster'
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: cassandra
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-couchbase-task
+spec:
+  sidecars:
+    - name: couchbase
+      image: couchbase/server-sandbox:5.5.0
+      readinessProbe:
+        httpGet:
+          path: /ui/index.html
+          port: 8091
+        # This Couchbase image recommends 60sec waiting for initial configuration
+        # Starting the tests too soon may result in
+        # "Error during initial configuration - aborting container"
+        # apparently because "vbucket map not available yet"
+        initialDelaySeconds: 60
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: couchbase
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-gevent-task
+spec:
+  params:
+  - name: imageTag
+    type: string
+  workspaces:
+    - name: task-pvc
+      mountPath: /workspace
+  steps:
+    - name: unittest
+      image: python:$(params.imageTag)
+      env:
+        - name: TEST_CONFIGURATION
+          value: gevent
+      workingDir: /workspace/python-sensor/
+      command:
+      - /workspace/python-sensor/.tekton/run_unittests.sh
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: python-tracer-unittest-default-task
+spec:
+  sidecars:
+    - name: google-cloud-pubsub
+      image: egymgmbh/pubsub-emulator
+      command:
+        - /init.sh
+        - test-project
+        - test-topic
+        - test-subscription
+    - name: mariadb
+      image: mariadb:11.2.3
+      env:
+        - name: MYSQL_ROOT_PASSWORD # or MARIADB_ROOT_PASSWORD
+          value: passw0rd
+        - name: MYSQL_DATABASE # or MARIADB_DATABASE
+          value: instana_test_db
+    - name: mongo
+      image: mongo:4.2.3
+    - name: postgres
+      image: postgres:9.6.24
+      env:
+        - name: POSTGRES_USER
+          value: root
+        - name: POSTGRES_PASSWORD
+          value: passw0rd
+        - name: POSTGRES_DB
+          value: instana_test_db
+      readinessProbe:
+        exec:
+          command:
+          - sh
+          - -c
+          - pg_isready --host 127.0.0.1 --port 5432 --dbname=${POSTGRES_DB}
+        timeoutSeconds: 10
+    - name: redis
+      image: redis:7.2.4
+    - name: rabbitmq
+      image: rabbitmq:3.12.12
```

As mentioned before, tracing against the most updated server versions would be good.

```suggestion
      image: rabbitmq
```